src/stateful-set.ts

```diff
@@ -178,21 +208,76 @@ export class StatefulSet extends workload.Workload implements IScalable {
 
   }
 
+  /**
+   * Add a PersistentVolumeClaim template to the StatefulSet.
+   *
+   * Each pod in the StatefulSet will get its own PVC based on this template.
+   *
+   * @param template The PVC template to add
+   */
+  public addVolumeClaimTemplate(template: PersistentVolumeClaimTemplate) {
+    this._volumeClaimTemplates.push(template);
+    return this;
+  }
+
+  /**
+   * Get all PersistentVolumeClaim templates that will be created for each pod.
+   */
+  public get volumeClaimTemplates(): PersistentVolumeClaimTemplate[] {
+    return [...this._volumeClaimTemplates];
+  }
+
   /**
     * @internal
     */
   public _toKube(): k8s.StatefulSetSpec {
+    // Get base pod spec
+    const basePodSpec = this._toPodSpec();
+
+    // Create a new pod spec for the StatefulSet with filtered volumes
+    const podSpec = { ...basePodSpec };
+
+    // When using volumeClaimTemplates, the volumes with matching names should not be included in the pod spec
+    if (this._volumeClaimTemplates.length > 0 && podSpec.volumes) {
+      const volumeClaimNames = this._volumeClaimTemplates.map(vct => vct.name);
+      const filteredVolumes = podSpec.volumes.filter(vol => !volumeClaimNames.includes(vol.name));
```

Comment:
Give more prescriptive names:

```suggestion
      const volumesWithNoTemplates = podSpec.volumes.filter(vol => !volumeClaimNames.includes(vol.name));
```